### PR TITLE
Various SDK related fixes

### PR DESCRIPTION
**Pull request by infomiho**
Fixes:
- `GitHub` import fixes
- Sendgrid `from from` fix
- Auth helpers are generated for GitHub and Google auth methods
- Exporting `Email` and `EmailFromField` types

### CODE REVIEW

File: waspc/data/Generator/templates/sdk/client/auth/github.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
 // PUBLIC API
-export { signInUrl as githubSignInUrl } from '../../auth/helpers/Github'
+export { signInUrl as githubSignInUrl } from '../../auth/helpers/GitHub'
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/client/auth/github.ts:2]: Fixes a typo

File: waspc/data/Generator/templates/sdk/client/auth/ui.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export type { CustomizationOptions } from '../../auth/forms/types'
 export { SignInButton as GoogleSignInButton } from '../../auth/helpers/Google'
 {=/ isGoogleAuthEnabled =}
 {=# isGithubAuthEnabled =}
-export { SignInButton as GithubSignInButton } from '../../auth/helpers/Github'
+export { SignInButton as GithubSignInButton } from '../../auth/helpers/GitHub'
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/client/auth/ui.ts:15]: Fixes a typo

File: waspc/data/Generator/templates/sdk/server/email/core/providers/sendgrid.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import SendGrid from "@sendgrid/mail";
 import { getDefaultFromField } from "../helpers.js";
-import type { SendGridProvider, EmailSender } from from "../types";
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/server/email/core/providers/sendgrid.ts:3]: Fixes a typo

File: waspc/data/Generator/templates/sdk/server/email/core/types.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ export type EmailSender = {
 // PRIVATE API
 export type SentMessageInfo = any;
 
-// PRIVATE API
+// PUBLIC API
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/server/email/core/types.ts:40]: Exposes `Email` as public api in `wasp/server/email` @Martinsos do we need to add it to the codemod?

File: waspc/data/Generator/templates/sdk/server/email/core/types.ts [hunk 5]
```diff
@@ -51,7 +51,7 @@ export type Email = {
   html: string;
 };
 
-// PRIVATE API
+// PUBLIC API
 export type EmailFromField = {
```
Comment (infomiho) [inline waspc/data/Generator/templates/sdk/server/email/core/types.ts:55]: Exposes `EmailFromField` in `wasp/server/email`

File: waspc/src/Wasp/Generator/SdkGenerator/AuthG.hs [hunk 6]
```diff
@@ -115,10 +115,7 @@ genUtils auth = return $ C.mkTmplFdWithData relUtilsFilePath tmplData
 
 genIndexTs :: AS.Auth.Auth -> Generator [FileDraft]
 genIndexTs auth =
-  return $
-    if isEmailAuthEnabled || isLocalAuthEnabled
```
Comment (infomiho) [inline waspc/src/Wasp/Generator/SdkGenerator/AuthG.hs:119]: `wasp/auth` was needed in the case of Github and Google auth as well, fixes the logic

File: waspc/data/Generator/templates/sdk/server/email/core/types.ts (hunk 4 above)
Comment (Martinsos) [reply waspc/data/Generator/templates/sdk/server/email/core/types.ts:40]: So you exposed `Email` and `EmailFromField` in this PR.

I don't think we ever had imports for these defined in the docs before, so, in theory we don't need them in the codemod? But if you think it was likely people used them before nevertheless, despite us not having them mentioned in the docs, then we can add them in the codemod. You can add them yourself every easily, just add the rule on the `main` branch -> if you are not sure about it, let me know and I can also do it or help.

**Review by Martinsos — APPROVED** [review]
Great @infomiho !

I left a comment regarding adding the email types to codemod.

File: waspc/data/Generator/templates/sdk/server/email/core/types.ts (hunk 4 above)
Comment (infomiho) [reply waspc/data/Generator/templates/sdk/server/email/core/types.ts:40]: Added them in case some people used it 👍

**Pull request merged**